packet/packet.go

```diff
@@ -51,27 +69,38 @@ func (packet *Packet) UnmarshalBinary(data []byte) error {
 
 // MarshalBinary implements the encoding.BinaryMarshaler interface
 func (packet *Packet) MarshalBinary() ([]byte, error) {
+	// form a buffer with the packet's magic code
 	buf := bytes.NewBuffer(packet.Code)
+
+	// write the request header
```

Comment:
Technically the magic code, the type, and the size all comprise the header, so this comment is a bit inaccurate.